qcodes_contrib_drivers/drivers/Aviosys/IP_Power_9258S.py

```diff
@@ -0,0 +1,92 @@
+from qcodes import Instrument, InstrumentChannel, ChannelList
+from qcodes.utils.validators import Enum
+from qcodes.utils.helpers import create_on_off_val_mapping
+import urllib.request
+
+
+class PowerChannel(InstrumentChannel):
+    """
+    Channel class for a socket on the Aviosys IP Power 9258S.
+
+    Args:
+        parent: Parent instrument.
+        name: Channel name.
+        channel: Alphabetic channel id.
+    """
+
+    CHANNEL_IDS = {'A': 1, 'B': 2, 'C': 3, 'D': 4}
+    CHANNEL_NAMES = Enum(*CHANNEL_IDS.keys())
+
+    def __init__(self, parent: Instrument, name: str, channel: str):
+
+        super().__init__(parent, name)
+
+        # validate the channel id
+        PowerChannel.CHANNEL_NAMES.validate(channel)
+        self._id_name = channel
+        self._id_number = PowerChannel.CHANNEL_IDS[channel]
+
+        # add parameters
+        self.add_parameter('power_enabled',
+                           get_cmd=self._get_power_enabled,
+                           set_cmd=self._set_power_enabled,
+                           get_parser=int,
+                           val_mapping=create_on_off_val_mapping(on_val="1", off_val="0"),
+                           label='Power {}'.format(self._id_name))
+
+    # get methods
+    def _get_power_enabled(self):
+        request = urllib.request.Request(self.parent.address+'/set.cmd?cmd=getpower')
+        response = urllib.request.urlopen(request)
+        request_read = response.read()
+        return request_read.decode("utf-8")[4 + self._id_number * 6]
+
+    # set methods
+    def _set_power_enabled(self, power):
+        request = urllib.request.Request(self.parent.address+'/set.cmd?cmd=setpower+p6%d=%s' % (self._id_number, power))
```

Comment:
I would consider using f string formatting here rather than the old style % formatting for readability but that is mostly a question of personal preference